File: Dec2020/2Dec/UniqueNumberOfOccurrences.java
/*
Problem Name: 1207. Unique Number of Occurrences
Problem Link: https://leetcode.com/problems/unique-number-of-occurrences/
*/
class Solution {
    public boolean uniqueOccurrences(int[] arr) {
        int n = arr.length;
        HashMap<Integer,Integer> map = new HashMap<Integer,Integer>();
        for(int i=0;i<n;i++)
            map.put(arr[i],map.getOrDefault(arr[i],0) + 1);
        Set<Integer> set = new HashSet<Integer>();
        set.addAll(map.values());
        return set.size() == map.size();
    }
}

File: Dec2020/2Dec/UniqueNumberOfOccurrences.cpp
/*
Problem name: 1207. Unique Number of Occurrences
Problem link: https://leetcode.com/problems/unique-number-of-occurrences/
*/

class Solution {
public:
    bool uniqueOccurrences(vector<int>& arr) {
        map<int,int> m;
        set<int> s;
        bool ans = true;
        for(int i=0;i<arr.size();++i){
            m[arr[i]]++;
        }
        for( auto l:m){
            if(s.find(l.second) == s.end()){
                s.insert(l.second);
            }
            else{
                ans = false;
                break;
            }
        }
        return ans;
    }
};
